src/AsyncStorage.native.js

```diff
@@ -322,6 +330,7 @@ const AsyncStorage = {
     callback?: ?(errors: ?$ReadOnlyArray<?Error>) => void,
   ): Promise<null> {
     return new Promise((resolve, reject) => {
+      checkValidArgs(arguments);
```

Comment:
I think it's unintuitive to check `arguments`. Can we instead explicitly check `keyValuePairs` and `callback`? For instance:

```js
if (!Array.isArray(keyValuePairs) ||
    keyValuePairs.length === 0 ||
    !Array.isArray(keyValuePairs[0]) {
  throw new Error("Expected key-value pairs as first argument to multiSet");
} else if (callback && typeof callback !== "function") {
  throw new Error("Expected function as second argument to multiSet");
}
```

Reply:
That's right about using `arguments. As for the errors, the issue I'm resolving is for cases when the array is not wrapped unintentionally. 
Although, the above error messages would cover that case, won't it be useful to have a `Did you mean to pass an array of key-value pairs instead?` message?

Reply:
Yeah, I'm all for adding more messages if it helps the author.

Reply:
Awesome! have made the changes. Please have a look 🙂